controllers/druid/handler.go

```diff
@@ -88,6 +89,45 @@ func deployDruidCluster(sdk client.Client, m *v1alpha1.Druid) error {
 		return err
 	}
 
+	/*
+		If EnablePVCDeletionFinalizer is set to false, finalizer shall not be executed
+		Default Behavior: Finalizer shall be always executed resulting in deletion of pvc post deletion of Druid CR
+		When the object (druid CR) has for deletion time stamp set, execute the finalizer
+		Finalizer shall execute the following flow :
+		1. Get sts List and PVC List
+		2. Range and Delete sts first and then delete pvc. PVC must be deleted after sts termination has been executed
+			else pvc finalizer shall block deletion since a pod/sts is referencing it.
+		3. Once delete is executed we block program and return.
+	*/
+
+	if m.Spec.DisablePVCDeletionFinalizer != true {
+		md := m.GetDeletionTimestamp() != nil
+		if md {
+			err := finalizer(sdk, m)
+			if err != nil {
+				return err
+			} else {
+				return nil
+			}
+		}
+		/*
+			If finalizer isn't present add it to object meta.
+			In case cr is already deleted do not call this function
+		*/
+		cr := checkIfCRExists(sdk, m)
```

Comment:
can we do this check after `ContainsString(m.ObjectMeta.Finalizers, finalizerName)` check so that we don't unnecessarily get the CR if finalizer is already there.

Reply:
ok so here's what i think
1. we check if CR exists
2. then check if that CR has string present or not
if we do the check after containerstring func that does not seems appropriate to me, since we must confirm if cr is present or removed
i hope thats why you meant right ? regardless open to suggestions :)

Reply:
ok, I see that method is also getting the current spec and not just checking the existence ...